atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -2214,7 +2215,21 @@ protected LockToken acquireLocksForCommit() {
                 lockDescriptors,
                 lockAcquireTimeoutMillis,
                 Optional.ofNullable(getStartTimestampAsClientDescription(currentTransactionConfig)));
-        LockResponse lockResponse = timelockService.lock(request);
+
+        RuntimeException stackTraceSnapshot = new RuntimeException("I exist to show you the stack trace");
```

Comment:
Creating a RE every time for the possible timeout is pretty smelly, but I don't really see a great way around it, because debugging this without a stacktrace would probably be a nightmare

Reply:
Yeah. I don't know: we could deal with some `StackTraceElement[]` objects instead, but I figured having an RE as a container for the stack trace was easier to use.

(Also, random thought: I'm guessing we pronounce RE differently, since I pronounce it as having a vowel sound as in arr-ih and I noticed you have an `a` as the article)

Reply:
That's because I don't read it as an acronym, just write it that way due to general laziness